src/clarity-angular/layout/vertical-nav/vertical-nav-group.ts

```diff
@@ -0,0 +1,186 @@
+import {style, animate, transition, state, trigger} from "@angular/animations";
+import {Subscription} from "rxjs";
+
+const EXPANDED_STATE: string = "expanded";
+const COLLAPSED_STATE: string = "collapsed";
+
+@Component({
+    selector: "clr-vertical-nav-group",
+    templateUrl: "./vertical-nav-group.html",
+    providers: [Expand],
+    animations: [
+        trigger("clrExpand", [
+            state(EXPANDED_STATE, style({
+                "height": "*",
+                "overflow-y": "hidden"
+            })),
+            state(COLLAPSED_STATE, style({
+                "height": 0,
+                "overflow-y": "hidden"
+            })),
+            transition("expanded <=> collapsed", animate("0.2s ease-in-out"))
+        ])
+    ],
+    host: {
+        "class": "nav-group"
+    }
+})
+export class VerticalNavGroup implements AfterContentInit, OnDestroy {
+
+    constructor(private _itemExpand: Expand,
+                private _verticalNavService: VerticalNavService) {
+        this._verticalNavService.registerNavGroup(this);
+
+
+        // FIXME: This subscription handles a corner case
+        // Vertical Nav collapse requires the animation to run first and then
+        // remove the nodes from the DOM. If the user directly sets the input
+        // on the clrIfExpanded directive, we have no chance to run the animation
+        // and wait for it to complete. This subscription makes sure that the
+        // animation states are correct for that edge case.
+        this._subscriptions.push(
+            this._itemExpand.expandChange.subscribe(value => {
+                    if (value && this.expandAnimationState === COLLAPSED_STATE) {
+                        if (this._verticalNavService.collapsed) {
+                            this._verticalNavService.collapsed = false;
+                        }
+                        this.expandAnimationState = EXPANDED_STATE;
+                    } else if (!value && this.expandAnimationState === EXPANDED_STATE) {
+                        this.expandAnimationState = COLLAPSED_STATE;
+                    }
+                }
+            )
+        );
+
+        // If the nav is collapsing, close the open nav group + save its state
+        // If the nav is expanding, expand if the previous state was expanding
+        this._subscriptions.push(
+            this._verticalNavService.animateOnCollapsed.subscribe((goingToCollapse: boolean) => {
+                    if (goingToCollapse && this.expanded) {
+                        this.wasExpanded = true;
+                        this.expandAnimationState = COLLAPSED_STATE;
+                    } else if (!goingToCollapse && this.wasExpanded) {
+                        this.expandGroup();
+                        this.wasExpanded = false;
+                    }
+                }
+            )
+        );
+    }
+
+    private wasExpanded: boolean = false;
+
+    @Input("clrVerticalNavGroupExpanded")
+    set userExpandedInput(value: boolean) {
```

Comment:
@youdz: I thought we wanted to support both with and without `clrIfExpanded` right? So the vertical nav still works if the user doesn't want to remove the links from the dom but they have to use this input for that. do you think i am missing something here?